src/data/markdown/docs/02 javascript api/07 k6-experimental/03 websockets/10 WebSocket/30-WebSocket-onmessage.md

```diff
@@ -0,0 +1,42 @@
+---
+title: 'WebSocket.onmessage(callback)'
+excerpt: 'Set up a callback function for message event WebSocket.'
+---
+
+Set a callback function as a handler for a WebSocket connection `message` event.
+For multiple, simultaneous event handlers, use [`WebSocket.addEventHandler()`](/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener).
+
+| Parameter | Type     | Description                                  |
+| --------- | -------- | -------------------------------------------- |
+| callback  | function | The function to call when the event happens. |
+
+### Example
+
+_A k6 script that initiates a WebSocket connection and sets up a handler for the `message` event._
+
+<CodeGroup labels={["example-websocket-onmessage.js"]} lineNumbers={[]} showCopyButton={[true]}>
+
+```javascript
+import { WebSocket } from 'k6/experimental/websockets';
+
+export default function () {
+  const ws = new WebSocket('ws://localhost:10000');
+
+  ws.onmessage = (data) => {
```

Comment:
I'm confused by the title of this page, `WebSocket.onmessage(callback)`, and this example. The title implies that `onmessage()` should be a function that receives a callback, but here we setting a function that receives the message data. I realize that this itself is the callback, but the way it's referred to in the title is confusing. The Parameter table is also confusing, since there's no `callback` parameter here.

For example, [MDN documents this](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/message_event) as being a handler for the `message` event. There's no mention of "callbacks". So I think we should also refer to these as event handlers instead.

Same applies for the other `.on*` pages.

Reply:
Sure, I'll change it to the handler everywhere  :+1: